packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { matchedSpecs } from '../../../src/sources'
+import { expect } from 'chai'
+
+describe('ProjectDataSource', () => {
```

Comment:
I think the test coverage should be expanded to cover more of what `transformSpec` does. For example, the handling of backslashes (`\`) in paths and the various spec files extensions.

Reply:
I added some more tests: https://github.com/cypress-io/cypress/pull/19319/commits/6cc7047e23157a26e4f9b1e14b1198c9bc1b4313

We seem to do a lot of checks around `\` all over the place, definitely want to consolidate this. Eg, it's not clear if we can rely on `projectRoot` to be posixified at this point, so for good measure I am normalizing the `\` to `/` here now.